### ISSUE TITLE
Inmediate overfitting when using Eager Execution

### ISSUE THREAD

**Issue by oscardp96** [tensorflow/tensorflow#29433]
**System information**
- OS Platform and Distribution: Google Colab in GPU mode
- TensorFlow version: 1.13.1
- Python version: Python 3.6.7

I am feeding a 3D Resnet with 3D images from a tfrecord file. Using exactly the same code, I obtain very different results when using Eager Execution. In fact, when I use eager execution, the model has 99% training accuracy since the first epoch, with just 20% validation accuracy. When not using eager execution, the behavior is completely normal, gradually incrementing accuracy with each epoch.

The 3D Resnet model I am using is from this rep: https://github.com/JihongJu/keras-resnet3d

**Code to reproduce the issue**
````
BATCH_SIZE = 8
STEPS_PER_EPOCH = int(n_training_samples / BATCH_SIZE)
VALIDATION_STEPS = int(n_val_samples / BATCH_SIZE)

image_tensor, label_tensor = dataset_parser(training_tfrec, BATCH_SIZE)
model = Resnet3DBuilder.build_resnet_18(
    input_shape=IMG_SHAPE,
    num_outputs=N_CLASSES,
    reg_factor=0.01
)

optimizer = keras.optimizers.Adam(lr=0.0001, decay=1e-6)
model.compile(optimizer=optimizer, 
              loss='categorical_crossentropy', 
              metrics=['acc'])

val_tensor = dataset_parser(validation_tfrec, BATCH_SIZE)

history = model.fit(x=image_tensor, y=label_tensor, 
                    epochs=10, steps_per_epoch=STEPS_PER_EPOCH, 
                    validation_data=val_tensor, validation_steps=VALIDATION_STEPS)

````

I realize this code is probably not enough, but maybe there are some clear reasons why this might be happening, and someone can help.

**Comment by ymodak** [comment]
Perhaps you can provide a complete code snippet and output for both versions, google colab gist will be great. Thanks!

**Comment by oscardp96** [comment]
I'm sorry but I was in a hurry and went back to graph execution to avoid the problem. Now I can't find the version of my Colab notebook where I had the code ready for eager execution. It would be nice to know what was happening, but I am afraid I cannot provide any more code. Thanks anyway.
